src/components/ICard/ICard.vue

```diff
@@ -27,13 +27,25 @@ export default defineComponent({
         size: {
             type: String,
             default: undefined
+        },
+        /**
+         * The HTML tag to render card as
+         * @default div
+         * @name tag
+         */
+        tag: {
+            type: String,
+            default: 'div'
         }
     },
     setup(props) {
         const currentColor = computed(() => props.color);
         const currentSize = computed(() => props.size);
         const { color } = useComponentColor({ componentName, currentColor });
         const { size } = useComponentSize({ componentName, currentSize });
+        const tagName = computed(() => {
+            return props.tag !== "div" ? props.tag : "article";
+        });
```

Comment:
This won't be needed. You can remove the computed tag and just use `:is="tag"`, since the prop defaults to `div`.